beetsplug/lastgenre/__init__.py

```diff
@@ -111,39 +111,41 @@ def setup(self):
             self.import_stages = [self.imported]
 
         self._genre_cache = {}
+        self.whitelist = self._load_whitelist()
+        self.c14n_branches, self.canonicalize = self._load_c14n_tree()
 
-        # Read the whitelist file if enabled.
-        self.whitelist = set()
+    def _load_whitelist(self) -> set[str]:
+        whitelist = set()
         wl_filename = self.config["whitelist"].get()
         if wl_filename in (True, ""):  # Indicates the default whitelist.
             wl_filename = WHITELIST
         if wl_filename:
             wl_filename = normpath(wl_filename)
             with open(wl_filename, "rb") as f:
-                for line in f:
-                    line = line.decode("utf-8").strip().lower()
+                for raw_line in f:
+                    line = raw_line.decode("utf-8").strip().lower()
```

Comment:
So I fixed the mypy errors I got:

```
beetsplug/lastgenre/__init__.py:126: error: Name "line" already defined on line 125  [no-redef]
beetsplug/lastgenre/__init__.py:127: error: Argument 1 to "startswith" of "bytes" has incompatible type "str"; expected "Buffer | tuple[Buffer, ...]"  [arg-type]
```

I need some advice here @snejus, @semohr, @wisp3rwind or @Serene-Arc : Is this the right way to do it? And isn't the loading as rb and then translating each line to str here anyway kind of redundant? Thanks in advance!

Reply:
ok I did my homework and researched about it. It's a good solution. Let's keep it. Resolved. Nevermind.